### PR TITLE
Documentation 

### PR DESCRIPTION
**Pull request by bibo7086**
I recently started work on the documentation for this project (mostly around the **`KademliaProtocol`** class) and have made some changes to enable the generation of javadoc documentation using Maven. This pull request fixes some errors that prevented the documentation from being generated correctly.

The changes include:
- Adding necessary plugins to the pom.xml file
- Editing code comments  to make them compatible with javadoc (inprogress)

### CODE REVIEW

File: simulator/src/main/java/peersim/kademlia/KademliaObserver.java [hunk 1]
```diff
@@ -92,27 +110,45 @@ private static void writeMap(Map<String, Map<String, Object>> map, String filena
...
+   */
   private static void writeMapFind(Map<String, Map<String, Object>> map, String filename) {
     try (FileWriter writer = new FileWriter(filename)) {
-      Set<String> keySet = map.entrySet().iterator().next().getValue().keySet();
-      String header = "";
-      for (Object key : keySet) {
-        header += key + ",";
-      }
+      // Define the expected order of keys in the header
+      String[] keys = {"src", "start", "stop", "messages", "hops", "id", "type"};
```
Comment (harnen) [inline simulator/src/main/java/peersim/kademlia/KademliaObserver.java:122]: Same here - let's read the keys from the map itself

File: simulator/src/main/java/peersim/kademlia/KademliaObserver.java [hunk 2]
```diff
@@ -55,33 +56,50 @@ public class KademliaObserver implements Control {
...
     try (FileWriter writer = new FileWriter(filename)) {
-      Set<String> keySet = map.entrySet().iterator().next().getValue().keySet();
-
-      String header = "";
-      for (Object key : keySet) {
-        header += key + ",";
-      }
+      // Set<String> keySet = map.entrySet().iterator().next().getValue().keySet();
+      // Define the expected order of keys in the header
+      String[] keys = {"src", "dst", "id", "type", "status"};
```
[the remark this answers is not in the document]
Comment (bibo7086) [reply simulator/src/main/java/peersim/kademlia/KademliaObserver.java]: Reverted it back to original implementation to handle any map input. Thanks

File: simulator/src/main/java/peersim/kademlia/KademliaObserver.java (hunk 1 above)
Comment (bibo7086) [reply simulator/src/main/java/peersim/kademlia/KademliaObserver.java]: Done

**Review by harnen — APPROVED** [review]
Looks good to me. Thanks for adding the documentation!

**Pull request merged**